### ISSUE TITLE
"Visible Time Range" doesn't work with transforms

### ISSUE THREAD

**Issue by Wumpf** [rerun-io/rerun#723]
**Describe the bug**
All transforms ignore the visible history field so far.

**To Reproduce**
It's best visible in the transforms3d demo of api_demo where everything on screen is logged only once but has changing transforms over time.

**Expected behavior**
For the demo in particular it's natural to expect traces of the planets as they go

**Additional context**
The underlying issue is that the transform cache doesn't know anything about visible history so far. Fixing this might be a bigger undertaking as it means that transforms shouldn't be queried & computed via the object path alone

**Comment by smileyenot983** [comment]
Is there any information about when this will be fixed?

**Comment by grtlr** [comment]
We just had this come up again, with more details (and a nice repro). The following might be helpful when fixing this:

* https://github.com/rerun-io/rerun/issues/10759#issue-3264695921

**Comment by soncininicolas** [comment]
This has been given less priority than it deserves I believe. It breaks with the "static=False" property whereby if I log a transform I expect it to stick to a given path and a given timeline, unless I specify the "static=True" property.

Also from a usability standpoint, it's useful to review the last N steps in any path taken by an entity (may be a camera, a robotic arm, a car, etc). I hit this issue when trying to visualize the last N camera poses in a given camera path (currently not possible).

I would love to help, though I'm not "fluent in Rust" :(

**Comment by Wumpf** [comment]
> It breaks with the "static=False" property whereby if I log a transform I expect it to stick to a given path and a given timeline, unless I specify the "static=True" property.

This sounds like a missunderstanding. `static` works as expected & intended already. What doesn't work is to visualize transforms accross a time range rather than the latest relative to the timecursor.

**Comment by mightymichael44** [comment]
Hey,
after being puzzled quite some time about blurred object history. I finally found this issue.
Didn't expect the transformations to not consider the timestamp of the data-to-be-transformed.
Is this feature/fix by chance planned within the next releases?

If possible I would ask to upvote the prio of this issue.

**Comment by Wumpf** [comment]
Sorry, unlikely to happen within the next few releases. It's not a priority for us right now and it's surprisingly tricky to make it useful & consistent: any change of any transform along a tree essentially re-instantiates everything down the tree which can lead to an explosion of things to draw for high frequency transforms. So we also likely need more ux around some kind of time sampling.
